### PR TITLE
#1376: Scalar - Replace all the IsEqual with ScalarHashValue

### PR DESCRIPTION
**Pull request by marceloamadeu**
This PR is for #1376

### CODE REVIEW

**Comment by marceloamadeu** [comment]
#1367 was removed because I didn't find any more cases of `IsEqual` that should be exchanged for `ScalarHashValue`.

To confirm, can you tell if in this case it is possible to use `ScalarHashValue`?

```
    public void dividesIntNumbers() {
        new Assertion<>(
            "Must divide int numbers",
            new DivisionOf(4, 2).intValue(),
            new IsEqual<>(2)
        ).affirm();
    }
```
Tks

**Comment by 0crat** [comment]
This pull request [#1383](https://github.com/yegor256/cactoos/issues/1383) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/07b82225-51cf-424a-aa90-803574137792); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/scalar/package-info.java [hunk 1]
```diff
@@ -29,9 +29,6 @@
  *  Assertion from cactoos-matchers. Once there is no more usage of
  *  MatcherAssert.assertThat, add the signature of MatcherAssert.assertThat to
  *  forbidden-apis.txt
- * @todo #1367:30min Replace all the IsEqual with ScalarHashValue in Scalar
```
Comment (victornoel) [inline src/test/java/org/cactoos/scalar/package-info.java:32]: @marceloamadeu have you verified there is really no IsEqual incorrectly used in the tests of `scalar` package?

**Comment by victornoel** [comment]
@marceloamadeu indeed, it seems to be ok then, thanks

@paulodamaso looks good to me

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@victornoel please make sure you will find at least 3 issues during next CR, thanks

**Comment by victornoel** [comment]
@sereshqua yes ^^ it's quite hard with this kind of PR :)

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**